scripts/west_commands/patch.py

```diff
@@ -0,0 +1,59 @@
+import argparse
+import os
+from pathlib import Path
+import subprocess
+
+from west import log
+from west.commands import WestCommand, CommandError
+from west.manifest import MANIFEST_PROJECT_INDEX
+from west.app.project import _ProjectCommand
+
+class Patch(_ProjectCommand):
+    def __init__(self):
+        super().__init__(
+            'patch',
+            'patch west subprojects',
+            'Runs "git am patches/<project>/" on each of the specified projects.')
```

Comment:
Can we add some additional help information? Like, where should patches be located? How should the patch files be named? What additional metadata should be present in the manifest?

Reply:
How about:
https://github.com/golioth/golioth-firmware-sdk/blob/6e9d68d4ee91225ac1977ec9f58c27b1304f255e/scripts/west_commands/patch.py#L9-L24